dsgrid/utils/dataset.py

```diff
@@ -22,14 +32,16 @@ def map_and_reduce_stacked_dimension(df, records, column, drop_column=True, to_c
         df = df.withColumn("fraction", F.lit(1.0))
     # map and consolidate from_fraction only
     records = records.filter("to_id IS NOT NULL")
-
-    df = df.join(records, on=df[column] == records.from_id, how="inner").drop("from_id")
+    df = df.join(records, on=getattr(df, column) == records.from_id, how="inner").drop("from_id")
     if drop_column:
         df = df.drop(column)
     df = df.withColumnRenamed("to_id", to_column_)
     nonfraction_cols = [x for x in df.columns if x not in {"fraction", "from_fraction"}]
-    df = df.fillna(1.0, subset=["from_fraction"]).selectExpr(
-        *nonfraction_cols, "fraction*from_fraction AS fraction"
+    # TODO DT: DuckDB doesn't have fillna. Do we even need it?
```

Comment:
Yes, no need to fillna() as we filter the records to to_id is NOT null in line 34.